components/world-4/breeding/pets.tsx

```diff
@@ -0,0 +1,532 @@
+                                        :
+                                        <Text margin={{top:'xsmall'}} size='small'>To avoid character checking for a global page, use this checkbox to consider it equipped or not</Text>
+                                    }
+                                </Box>
+                            }
+                        >
+                            <CircleInformation size="small" />
+                        </TipDisplay>
+                    </Box>
+                    <Box direction='row' gap='xsmall'>
+                        <CheckBox
+                            checked={silkRodeChip}
+                            label="Silkrode Nanochip Equipped"
+                            onChange={(event) => setSilkrode(event.target.checked)}
+                            disabled={!starSignEquipped}
+                        />
+                        <TipDisplay
+                            heading="Silkrode Nanochip"
+                            size='medium'
+                            maxWidth='medium'
+                            body={
+                                <Box>
+                                    <Text size='small'>You can check this checkbox to get accurate values when equipping the Silkrode Nanochip in the Lab (double star sign bonus)</Text>
+                                </Box>
+                            }
+                        >
+                            <CircleInformation size="small" />
+                        </TipDisplay>
+                    </Box>
+                </Box>
+            }
+            </Box>
+            {activeTab == "All" && <AllPetDisplay />}
+            {activeTab == "Shiny" && <ShinyDisplay silkRodeChip={silkRodeChip} starSignEquipped={starSignEquipped} />}
+            {activeTab == "Breedability" && <BreedabilityDisplay silkRodeChip={silkRodeChip} starSignEquipped={starSignEquipped} />}
+        </Box>
+    )
+}
+
+function AllPetDisplay() {
+    const { theData } = useAppDataStore(useShallow(
+        (state) => ({ theData: state.data.getData(), lastUpdated: state.lastUpdated })
+    ));
+
+    // Get breeding data, if it's not available yet just show placeholder loading.
+    const breeding = theData.get("breeding") as BreedingDomain;
+
+    if (!breeding) {
+        return (
+            <Box>
+                Still loading...
+            </Box>
+        )
+    }
+
+    // Will need to update this once new pets from world 5 are implemented into the game
+    const worldsToDisplay = [0,1,2,3];
+
+    return (
+        worldsToDisplay.map(world => {
+            // Get all the pets for this world, and sort them in the unlock order
+            const petsOfWorld = breeding.basePets.filter(pet => pet.data.world == world).slice().sort((pet1, pet2) => pet1.data.unlockOrder > pet2.data.unlockOrder ? 1 : -1);
```

Comment:
Maybe better to move this into a `useMemo`? It won't change without data change so doesnt' really need to be inside the rendering section, it's pretty harmless so doesn't have to happen but just a thought.

Reply:
To be honest I feel like it cleaner to have it rendered like that directly, instead of creating some arrays that will need to be parsed anyway, but maybe there's a clean way to do it in a way I'm not thinking about

Reply:
To clarify what I meant, you could do something like this (I wrote this in git so maybe syntax is wrong lol):
```
const petsOfWorld = useMemo(() => (
   breeding.basePets.filter(pet => pet.data.world == world).slice().sort((pet1, pet2) => pet1.data.unlockOrder > pet2.data.unlockOrder ? 1 : -1);
), [breeding]);
```

And you place it under `const worldsToDisplay = [0,1,2,3];` (i.e. outside of the `return` statement).

That way this filter won't be re-calculated everytime you change tabs but only when the data reloads.

Hopefully that makes more sense, if not you can ignore this as I said.

Reply:
I get what you mean, I've done a commit that should be more to your linking for this